backend/amd64/proc.ml

```diff
@@ -146,17 +146,15 @@ let hard_float_reg =
 let all_phys_regs =
   Array.append hard_int_reg hard_float_reg
 
-let phys_reg n =
-  if n < 100 then hard_int_reg.(n) else
-  if n < 200 then hard_float_reg.(n - 100)
-  else Misc.fatal_errorf "Register of unknown class (%d)" n
+let phys_reg ty n =
+  Reg.at_location ty (Reg n)
```

Comment:
I think we should not create a new one every time, preallocate them like before, but return by type. See comment in `Reg.reset`.

Reply:
Now preallocated